src/main/java/ch/njol/skript/command/Commands.java

```diff
@@ -268,19 +268,26 @@ static boolean handleEffectCommand(final CommandSender sender, String command) {
 			command = "" + command.substring(SkriptConfig.effectCommandToken.value().length()).trim();
 			final RetainingLogHandler log = SkriptLogger.startRetainingLog();
 			try {
+				// Call the event on the Bukkit API for addon developers.
+				EffectCommandEvent effectCommand = new EffectCommandEvent(sender, command);
+				Bukkit.getPluginManager().callEvent(effectCommand);
+				command = effectCommand.getCommand();
 				ParserInstance parserInstance = ParserInstance.get();
 				parserInstance.setCurrentEvent("effect command", EffectCommandEvent.class);
-				Effect e = Effect.parse(command, null);
+				Effect effect = Effect.parse(command, null);
 				parserInstance.deleteCurrentEvent();
 				
-				if (e != null) {
+				if (effect != null) {
 					log.clear(); // ignore warnings and stuff
 					log.printLog();
-					
-					sender.sendMessage(ChatColor.GRAY + "executing '" + ChatColor.stripColor(command) + "'");
-					if (SkriptConfig.logPlayerCommands.value() && !(sender instanceof ConsoleCommandSender))
-						Skript.info(sender.getName() + " issued effect command: " + command);
-					TriggerItem.walk(e, new EffectCommandEvent(sender, command));
+					if (!effectCommand.isCancelled()) {
+						sender.sendMessage(ChatColor.GRAY + "executing '" + ChatColor.stripColor(command) + "'");
+						if (SkriptConfig.logPlayerCommands.value() && !(sender instanceof ConsoleCommandSender))
+							Skript.info(sender.getName() + " issued effect command: " + command);
+						TriggerItem.walk(effect, effectCommand);
```

Comment:
Could you insert this line after the walking: `Variables#removeLocals(effectCommand)`?
Otherwise we might get memory leaks again, though unlikely because effect commands are just a single effect